modules/db/src/main/scala/gem/dao/StepDao.scala

```diff
@@ -240,11 +240,19 @@ object StepDao {
     * exception.
     *
     * @param oid observation whose step configurations are sought
-    * @param i instrument associated with the observation
     */
-  def selectAll(oid: Observation.Id, i: Instrument): ConnectionIO[List[Step[InstrumentConfig]]] =
-    (i match {
-      case Instrument.Flamingos2 => selectAllF2(oid)
-      case _                     => selectAllGeneric(oid)
-    }).map(_.map(_.widen[InstrumentConfig]))
+  def selectAll(oid: Observation.Id): ConnectionIO[List[Step[InstrumentConfig]]] = {
+    def instrumentConfig(ss: List[Step[Instrument]]): ConnectionIO[List[InstrumentConfig]] =
+      ss.headOption.fold(List.empty[InstrumentConfig].point[ConnectionIO]) { s =>
+        (s.instrument match {
+          case Instrument.Flamingos2 => selectF2(oid)
+          case _                     => selectGeneric(oid)
+        }).map(_.map(c => c: InstrumentConfig))  // List isn't a scalaz Functor? So no widen on List?
+      }
```

Comment:
The snippet comes from a shell command that starts up the REPL with scalaz.   It was using scalaz 7.1.1 but when updated to 7.2.4 `widen` starts working.

Now, as long as I apply it to every branch of the `match` it works:

````scala
    ss.headOption.fold(List.empty[InstrumentConfig].point[ConnectionIO]) { s =>
      s.instrument match {
        case Instrument.Flamingos2 => selectF2(oid).map(_.widen[InstrumentConfig])
        case _                     => selectGeneric(oid).map(_.widen[InstrumentConfig])
      }
    }
````

but if I try to map the result to a widened list the compiler complains:

````
value widen is not a member of _187
}).map(lst => lst.widen[InstrumentConfig])
                  ^
````

Presumably since `List` is covariant there's not even really a need to explicitly widen.  This awkwardness works:

````scala
      ss.headOption.fold(List.empty[InstrumentConfig].point[ConnectionIO]) { s =>
        (s.instrument match {
          case Instrument.Flamingos2 => selectF2(oid)
          case _                     => selectGeneric(oid)
        }).map((lst: List[InstrumentConfig]) => lst)
      }

```` 

Finally, the `foldMap` version also works with a few adjustments but doesn't seem much better, if I'm doing it right:

````scala
      ss.headOption.map(_.instrument).foldMap[ConnectionIO[List[InstrumentConfig]]] {
        case Instrument.Flamingos2 => selectF2(oid).map(_.widen[InstrumentConfig])
        case _                     => selectGeneric(oid).map(_.widen[InstrumentConfig])
      }
````

So after all of that, what's the proper way to do this?